src/auth/strategies/rt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { Request } from 'express';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { JwtPayload, JwtPayloadWithRT } from 'src/@types/auth';
+
+@Injectable()
+export class JwtRtStrategy extends PassportStrategy(Strategy, 'jwt-refresh') {
+  constructor() {
+    super({
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+      secretOrKey: process.env.JWT_SECRET,
+      passReqToCallback: true,
+    });
+  }
+
+  validate(req: Request, payload: JwtPayload): JwtPayloadWithRT {
+    const refreshToken = req
+      .get('authorization')
+      ?.replace('Bearer', '')
+      ?.trim();
+
+    return {
+      ...payload,
+      refreshToken,
```

Comment:
It's returning `undefined` if `authorization` is not in request.
Did you mean it?

Reply:
nvm if u meant it